cloudinit/sources/DataSourceAzure.py

```diff
@@ -2203,7 +2217,7 @@ def _generate_network_config_from_imds_metadata(imds_metadata) -> dict:
                 privateIp = addr["privateIpAddress"]
                 if not dev_config.get("addresses"):
                     dev_config["addresses"] = []
-                dev_config["addresses"].append(
+                dev_config["addresses"].append(  # type: ignore
```

Comment:
Hmm, using pylance these 3 in this section don't trigger anything. Curious what mypy doesn't like.

Reply:
cloudinit/sources/DataSourceAzure.py:2217: error: "object" has no attribute "append"

Reply:
pyright makes no complaints about it either

Reply:
Setting the dictionaries to Dict[str, Any] seems to do the trick as well for these :D